agent_monitor/src/main.cpp

```diff
@@ -379,6 +381,213 @@ std::map<std::string, double> computeDeltaDist(std::map<std::string, TRBuffer <
     return deltaDistMap;
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/**
+* @brief This function compute the map required by the fact "IsLookingToward" 
+*        by testing if an entity is lying in the 3D cone of agent visual attention
+* @param Entity map containing all entity involved in the joint action & their 
+*        respective ids
+* @param Monitored agent id
+* @param Distance beetween center of basement circle and agent head position
+* @param Angular aperture of the cone in radians
+* @return Map required by the fact "IsLookingToward" containing all entities 
+*         lying in the cone and all normalized angles beetween entities and cone axis         
+*/
+//for convenience
+typedef std::vector<double> Vec_t;
+typedef std::vector<Vec_t> Mat_t;
+typedef std::map<std::string, double> Map_t;
+typedef std::pair<std::string, double> Pair_t;
+
+Vec_t operator*(const Mat_t &a, const Vec_t &x)
+{
+  int i,j;
+  int m = a.size();
+  int n = x.size();
+
+  Vec_t prod(m);
+
+  for(i = 0; i < m; i++){
+    prod[i] = 0.;
+    for(j = 0; j < n; j++)
+      prod[i] += a[i][j]*x[j];
+  }
+  return prod;
+}
+
+Vec_t operator-(const Vec_t &a , const Vec_t &b)
```

Comment:
I think this kind of basic computation function should be in MathFunction in toasterlib.
Can you do that?

I don't want the main to be too long, plus these kind of functions could be usefull in other modules